webdriver/tests/bidi/script/evaluate/serialization.py

```diff
@@ -28,3 +28,96 @@ async def test_primitive_values(bidi_session, top_context, expression, expected)
     )
 
     assert result == expected
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "expression, expected",
+    [
+        (
+            "new RegExp(/foo/)",
+            {
+                "type": "regexp",
+                "value": {
+                    "pattern": "foo",
+                    "flags": "",
+                },
+            },
+        ),
+        (
+            "new Date(1654004849000)",
+            {
+                "type": "date",
+                "value": "2022-05-31T13:47:29.000Z",
+            },
+        ),
+        (
+            "[1, '2', true, new RegExp(/foo/g), [1]]",
+            {
+                "type": "array",
+                "value": [
+                    {"type": "number", "value": 1},
+                    {"type": "string", "value": "2"},
+                    {"type": "boolean", "value": True},
+                    {
+                        "type": "regexp",
+                        "value": {
+                            "pattern": "foo",
+                            "flags": "g",
+                        },
+                    },
+                    {"type": "array"},
+                ],
+            },
+        ),
+        (
+            "new Map([[1, 2], ['2', '3'], [true, false], ['4', [1]]])",
+            {
+                "type": "map",
+                "value": [
+                    [
+                        {"type": "number", "value": 1},
+                        {"type": "number", "value": 2},
+                    ],
+                    ["2", {"type": "string", "value": "3"}],
+                    [
+                        {"type": "boolean", "value": True},
+                        {"type": "boolean", "value": False},
+                    ],
+                    ["4", {"type": "array"}],
+                ],
+            },
+        ),
+        (
+            "new Set([1, '2', true, [1]])",
+            {
+                "type": "set",
+                "value": [
+                    {"type": "number", "value": 1},
+                    {"type": "string", "value": "2"},
+                    {"type": "boolean", "value": True},
+                    {"type": "array"},
+                ],
+            },
+        ),
+        (
+            "{'foo': {'bar': 'baz'}, 'qux': 'quux'}",
```

Comment:
On Firefox this is a syntax error. As an expression, the opening and closing braces create a block, not an object.
We should wrap it with parens. 
```suggestion
            "({'foo': {'bar': 'baz'}, 'qux': 'quux'})",
```

Reply:
done ✅